packages/react/utils.ts

```diff
@@ -22,14 +24,26 @@ export const unwrap = (vnode?: ReactNode): VNode => {
       unwrap(child),
     );
   }
+
   return {
-    type: vnode.type, // lets pretend no function go through
+    type, // lets pretend no function go through
     props,
   };
 };
 
 export const flatten = <T>(rawChildren: T): T[] => {
   if (rawChildren === undefined || rawChildren === null) return [];
+  if (
+    typeof rawChildren === 'object' &&
+    'type' in rawChildren &&
+    'props' in rawChildren &&
+    rawChildren.props &&
+    typeof rawChildren.props === 'object' &&
+    'children' in rawChildren.props &&
+    rawChildren.type === FRAGMENT_SYMBOL
```

Comment:
Do we need this many checks? We could only check the `type` property (and if `rawChildren` is an object) imo

Reply:
Any updates on this?

Reply:
fixed, removed type generic